File: Decorator/Java/DecoratorCoffee/src/BaseDecorator.java
public class BaseDecorator implements ICoffee{
    protected ICoffee coffee;

    public BaseDecorator(ICoffee coffee) {
        this.coffee = coffee;
    }

    @Override
    public double getCost() {
        return coffee.getCost();
    }

    @Override
    public String getDescription() {
        return coffee.getDescription();
    }
}

File: Decorator/DecoratorCoffee/DecoratorCoffee/BaseDecorator.cs
namespace DecoratorCoffee
{
    public class BaseDecorator : ICoffee
    {
        protected ICoffee _coffee;

        public BaseDecorator(ICoffee coffee)
        {
            _coffee = coffee;
        }

        public virtual double GetCost() => _coffee.GetCost();
        public virtual string GetDescription() => _coffee.GetDescription();
    }
}
